app/auth.go

```diff
@@ -676,7 +676,14 @@ func (m *menderAuthManagerService) MakeAuthRequest() (*client.AuthRequest, error
 
 	tentok := strings.TrimSpace(string(m.tenantToken))
 
-	log.Debugf("Tenant token: %s", tentok)
+	tentok4Log := tentok
+
+	// Truncates the tenant token for avoid any leak
+	if len(tentok) > 4 {
+		tentok4Log = tentok[0:2] + "..." + tentok[len(tentok)-2:]
+	}
+
+	log.Debugf("Tenant token: %s", tentok4Log)
```

Comment:
With regards to the general logic. It won't happen in practice, but look at the following:
* do a copy of tenok to tenok4log
* `if` something something, then redact tenok4log for sercure logging
* log unconditionally

See the issue? If the `if` is false then you log the original secret tenok.

There are many ways to solve this. One way:
```suggestion
	tentok4Log := ""

	// Truncates the tenant token for avoid any leak
	if len(tentok) > 4 {
		tentok4Log = tentok[0:2] + "..." + tentok[len(tentok)-2:]
	}

	log.Debugf("Tenant token: %s", tentok4Log)
```

and a simpler one:

```suggestion
	// Truncates the tenant token for avoid any leak
	if len(tentok) > 4 {
		log.Debugf("Tenant token: %s..%s", tentok[0:2], tentok[len(tentok)-2:])
	}
```